### PR TITLE
Update the d3 to version7

### LINKED ISSUE #1367: Update `d3`
Upgrading `d3` is a significant piece still to be done, and which I think we'll tackle separately. We're behind two major releases and the `d3-tip` companion library seems to have been archived on GitHub (https://github.com/caged/d3-tip). I found a migration guide for version 6, but no equivalent yet for version 7: https://observablehq.com/@d3/d3v6-migration-guide

_Originally posted by @ssciolla in https://github.com/tl-its-umich-edu/my-learning-analytics/pull/1363#discussion_r844336733_

As of now (4/8/22), I don't think `d3` version 5 poses any vulnerabilities, but updating it is something we'll have to face at some point.

### PR DESCRIPTION
**Pull request by pushyamig**
Fixes #1367 .

Update to d3-v7 is a breaking change and needed an update to all 3 visuals ( Resource accessed, Grade distribution, Old assignment planning)

- The main change is to the d3.event is deleted but can be accessed as part of the JS events .https://github.com/d3/d3/blob/main/CHANGES.md
- Migration guide to [v6](https://observablehq.com/@d3/d3v6-migration-guide) . I did not see a whole lot of difference to v7 affecting us 
- Updating the [d3-tip](https://github.com/caged/d3-tip) to new library [d3-v6-tip](https://github.com/bumbeishvili/d3-v6-tip#history).  
      - The [new library](https://github.com/bumbeishvili/d3-v6-tip#foreword) is based on the old library
      -  I don't see the d3-v6-tip updated to v7 but seems to be working fine.
- Assignment planning old needed some changes since it was breaking.
- New assignment planning don't use D3

### CODE REVIEW

File: assets/src/components/TableAssignment.js [hunk 1]
```diff
@@ -1,17 +1,16 @@
 // modified from https://demos.creative-tim.com/material-dashboard-react/?_ga=2.12819711.913135977.1549993496-494583875.1549993496#/table
```
Comment (pushyamig) [inline assets/src/components/TableAssignment.js:2]: Old assignment planning change. I just did what worked.

File: assets/src/components/Tooltip.js [hunk 2]
```diff
@@ -1,10 +1,11 @@
 import React from 'react'
```
Comment (pushyamig) [inline assets/src/components/Tooltip.js:1]: Old assignment planning change. I just did what worked.

File: assets/src/components/d3/createProgressBar.js [hunk 3]
```diff
@@ -56,8 +56,8 @@ function createProgressBar ({ data, width, height, domElement, tip }) {
     .style('outline-color', 'rgb(255, 255, 255)')
```
Comment (pushyamig) [inline assets/src/components/d3/createProgressBar.js:56]: Old assignment planning change. I just did what worked.

File: assets/src/util/createToolTip.js [hunk 4]
```diff
@@ -1,6 +0,0 @@
-import d3tip from 'd3-tip'
```
Comment (pushyamig) [inline assets/src/util/createToolTip.js:1]: Old assignment planning change. I just did what worked.

File: assets/src/components/Tooltip.js (hunk 2 above)
Comment (pushyamig) [reply assets/src/components/Tooltip.js:1]: This change was made due since the event is passed as one of the arguments and cannot be access as `d3.event`

More on this https://observablehq.com/@d3/d3v6-migration-guide#events 

This change applies for all assignment and resource accessed view

File: assets/src/components/d3/createResourceAccessChart.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import * as d3 from 'd3'
```
Comment (pushyamig) [inline assets/src/components/d3/createResourceAccessChart.js:1]: This change was made due since the event is passed as one of the arguments and cannot be accessed as `d3.event`

More on this https://observablehq.com/@d3/d3v6-migration-guide#events

Comment (pushyamig) [reply assets/src/components/d3/createResourceAccessChart.js:1]: Lots of warnings were thrown during the `docker compose up ` w.r.t to `d3.event`. So fixes were based on those alerts and the migration guide help

File: assets/src/components/d3/createProgressBar.js (hunk 3 above)
Comment (ssciolla) [reply assets/src/components/d3/createProgressBar.js:56]: I'm seeing a console error sometimes that originates on line 50:
```
 Error: <rect> attribute width: Expected length, "NaN".
```

If you change the `attr('width...` part to this, you can see sometimes `NaN` is returned:
```
    .attr('width', d => {
      const result = x(d.percent_gotten) - margin.left
      console.log(result)
      return result
    })
```

File: assets/src/components/d3/createResourceAccessChart.js [hunk 6]
```diff
@@ -250,17 +251,17 @@ function createResourceAccessChart ({ data, weekRange, gradeSelection, resourceT
...
+  function brushmove (event) {
     const fullRange = mainYZoom.range()
-    const selection = d3.event
-      ? d3.event.selection[1] === 0 // prevents [0, 0] from being returned, which causes bug
-        ? [0, 0.1]
-        : d3.event.selection
+    const selection = event
+      ? event.selection[1] === 0 // prevents [0, 0] from being returned, which causes bug
+          ? [0, 0.1]
+          : event.selection
```
Comment (ssciolla) [inline assets/src/components/d3/createResourceAccessChart.js:264]: Nit-pick: check whether indentation is correct.

**Review by ssciolla — COMMENTED** [review]
This seems to be working for the most part. I'm flagging one error that I saw in old Assignment Planning, but I don't actually know if it's the result of the version change or not. I'll finish reviewing next week.

File: assets/src/components/d3/createProgressBar.js (hunk 3 above)
Comment (pushyamig) [reply assets/src/components/d3/createProgressBar.js:56]: I got a fix for this and I discussed this with @ssciolla

File: assets/src/components/Tooltip.js [hunk 7]
```diff
@@ -1,10 +1,11 @@
...
 import { renderToString } from 'react-dom/server'
-import createToolTip from '../util/createToolTip'
 import Paper from '@material-ui/core/Paper'
 import Typography from '@material-ui/core/Typography'
+import { tip as d3tip } from 'd3-v6-tip'
 
-export const AssignmentPlanningTooltip = classes => createToolTip(d => renderToString(
+export const AssignmentPlanningTooltip = classes => d3tip().html((event, d) => {
+  return renderToString(
```
[the remark this answers is not in the document]
Comment (pushyamig) [reply assets/src/components/Tooltip.js]: I think I indented as suggested, but I have a different linter so I am not sure if it same as your.

File: assets/src/components/d3/createResourceAccessChart.js (hunk 6 above)
Comment (pushyamig) [reply assets/src/components/d3/createResourceAccessChart.js]: I think I indented as suggested, but I have a different linter so I am not sure if it same as your.

File: assets/src/components/d3/createProgressBar.js (hunk 3 above)
Comment (ssciolla) [reply assets/src/components/d3/createProgressBar.js:56]: No longer seeing the error 👍

File: assets/src/components/d3/createResourceAccessChart.js (hunk 6 above)
Comment (ssciolla) [reply assets/src/components/d3/createResourceAccessChart.js]: Looks good.

**Review by ssciolla — APPROVED** [review]
I think this looks good. We'll need to do some some pretty thorough testing, hopefully with a range of courses to make sure no other edge cases cause problems and looking at the browser console.
